content/PGP/YubiKey 5.2.3 Enhancements to OpenPGP 3.4.adoc

```diff
@@ -0,0 +1,168 @@
+
+Current stable releases (2.2.12) of gnupg support enabling KDF, but do not have any indication when it is enabled. This will be added in the upcoming GnuPG 2.3.0, which displays KDF state as part of the card status.
+
+* Enable KDF
+[source, python]
+-----------------
+$ gpg --edit-card
+gpg/card> admin
+gpg/card> kdf-setup
+gpg/card> passwd
+gpg: OpenPGP card no. D2760001240103030006294453760000 detected
+ 
+1 - change PIN
+2 - unblock PIN
+3 - change Admin PIN
+4 - set the Reset Code
+Q - quit
+ 
+Your selection? 1
+PIN changed.
+-----------------
+
+=== Multiple cardholder certificate slots 
+With OpenPGP 3.4, support for additional X.509 certificates for the Authentication, Signing and Decipher keys has been added. Where as previously, these keys were treated as sub-keys for the OpenPGP certificate, now they can behave as independently issued credentials assigned to the user.
+
+The YubiKey (5.2.3 and above) supports 2KB per key certificate for up to 4 slots:
+
+* Authentication
+* Decryption
+* Signature
+* Attestation template
+
+Slots can be iterated over with the GET_NEXT_DATA instruction. These certificates may be used to identify the card in a client-server authentication, where specific non-OpenPGP certificates are needed, such as S-MIME and other x.509 related functions. The certificates are stored in the following order: Authentication, Decipher, Signature and Attestation. 
+
+GnuPG does not currently support accessing all certificates.  The authentication certificate can be read with:
+
+[source, python]
+-----------------
+$ gpg --edit-card
+...
+gpg/card> admin
+gpg/card> readcert 3 > aut_cert.der
+gpg/card> writecert 3 < aut_cert.der
+-----------------
+Reading the other slots is not currently possible with gpg 2.2.12. However, the YubiKey Manager can be used to load and manage the cardholder certificates for the Authentication, Decipher and Signing.
+
+
+=== Touch Cache
+The YubiKey has implemented support for the touch-to-verify function with OpenPGP for existing and prior firmware, allowing users to require any OpenPGP cryptographic event to be verified with a user’s touch on the YubiKey hardware before proceeding. With the release of the 5.2.3 firmware, this function has been extended to support a “Touch Cache” feature. With this option enabled, the user verification touch event unlocks the OpenPGP cryptographic functions for up to 15 seconds or until the OpenPGP session ends. This allows for subsequent authentication or signing events to be validated with a single touch, enhancing the user experience. The Touch Cache options can be set using the https://www.google.com/url?q=https://www.yubico.com/products/services-software/download/yubikey-manager/&sa=D&ust=1566260477104000[YubiKey Manager].
+
+=== General Enhancements
+The OpenPGP Smart card 3.4 specifications include a number of enhancements designed to improve general functionality and speed. These include:
+
+* Expanded Length Responses

```

Comment:
Should probably replace "Expanded" with "Extended", since that's the official term in the spec